pkg/operator/ceph/object/objectstore.go

```diff
@@ -591,33 +590,41 @@ func deletePools(context *Context, spec cephv1.ObjectStoreSpec, lastStore bool)
 		pools = append(pools, rootPool)
 	}
 
-	var wg sync.WaitGroup
-	for _, pool := range pools {
-		wg.Add(1)
-		name := poolName(context.Name, pool)
-		go func() {
-			defer wg.Done()
-			if err := cephclient.DeletePool(context.Context, context.clusterInfo, name); err != nil {
+	if configurePoolsConcurrently() {
+		waitGroup, _ := errgroup.WithContext(context.TODO())
+		for _, pool := range pools {
+			name := poolName(ctx.Name, pool)
+			waitGroup.Go(func() error {
+				if err := cephclient.DeletePool(ctx.Context, ctx.clusterInfo, name); err != nil {
+					logger.Warningf("failed to delete pool %q. %v", name, err)
+				}
+				return nil
+			},
+			)
+		}
+
+		// Wait for all the pools to be deleted
+		waitGroup.Wait()
```

Comment:
If the error is returned from the waitgroup, then you could log the warning here
```suggestion
		if err := waitGroup.Wait(); err != nil {
			logger.Warning(err)
		}
```